src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -146,7 +146,7 @@ def _search_role_definitions(cli_ctx, definitions_client, name, scopes, custom_r
 
 def create_role_assignment(cmd, role, assignee=None, assignee_object_id=None, resource_group_name=None,
                            scope=None, assignee_principal_type=None, description=None,
-                           condition=None, condition_version=None):
+                           condition=None, condition_version=None, assignment_name=None):
```

Comment:
`assignment_name` is not required, but optional. If we use code-gen for it, this certainly needs special handling.